api/v1beta1/metallb_types.go

```diff
@@ -55,6 +55,27 @@ type MetalLBSpec struct {
 	// +optional
 	// +kubebuilder:validation:Enum=all;debug;info;warn;error;none
 	LogLevel MetalLBLogLevel `json:"logLevel,omitempty"`
+
+	// Definie priority class name for MetalLB pods.
+	// +optional
+	PriorityClassName string `json:"priorityClassName,omitempty"`
```

Comment:
I would differentiate between speaker and controller here, reasonably with a sub structure.
Also, the node selector and tolerations must be added to the controller too.

Reply:
good thought ! added sub structure without breaking backward compatibility.